containers/java-postgres/test-project/src/test/java/com/mycompany/app/AppTest.java

```diff
@@ -0,0 +1,86 @@
+/*--------------------------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+ *-------------------------------------------------------------------------------------------------------------*/
+
+package com.mycompany.app;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import java.net.InetAddress;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+public class AppTest
+{
+    private Connection CreateConnection(String host, String username, String password) throws Exception{
+        Connection c = null;
+        Class.forName("org.postgresql.Driver");
+        c = DriverManager
+            .getConnection("jdbc:postgresql://" + host + "/postgres",
+            username, password);
+        return c;
+    }
+            
+    public AppTest() {
+    }
+
+    @Test
+    public void testApp()
+    {
+        assertTrue( true );
+    }
+
+    @Test
+    public void testMore()
+    {
+        assertTrue( true );
+    }
+
+    @Test
+    public void testIP() throws Exception
+    {
+        String host = System.getenv("POSTGRES_HOSTNAME");
+
+        InetAddress postgresAddress = InetAddress.getByName(host);
+        System.out.println("Sending Ping Request to " + host);
+        if (postgresAddress.isReachable(5000)){
+            System.out.println("Successfully Reached: " + host);
+            return;
+        }
+        System.out.println("Could not reach or connect to: " + host);
```

Comment:
qq from my limited Java knowledge: How does this test `pass/fail` if either:
- There isn't an assertion statement on the host being reachable,
- or there is no exception being explicitly thrown if we cannot connect to the host.

Reply:
@Kpopovs1, the test method is throwing the generic exception (look at the signature), so whatever method call fails inside the test, it is reported as an error.

Reply:
Oh ok, but @Kpopovs1 , you're right. line 55 should be an assertFalse, since it could be the case that it is not reachable, but it doesn't timeout.
@dnastrain , @KanishkT123 , seems like that is missing here on line 55

Reply:
hi @Kpopovs1, the code in place is actually what we landed on upon reviewing it with @juzuluag, where we intentionally are letting the maven+junit test runner interpret test-case exceptions as failed-test results, rather than catch them and then 're-raise a failure' to junit -- and testing w/ juzuluag we also confirmed that exceptions are thrown (by java) upon connectivity failures, which are correctly reported as 'test case failures' by `./test.sh`